### PR TITLE
187-No-login?-no-sensible-data?

### LINKED ISSUE #187: No login? No sensible data!
We have 3 kinds of user in our system. 
Normalos (login, openid connect: students, profs, ...), Admins (extra admin features), Strangers (login skipped). 
We welcome unknowns, but we want them to be able to search only publicly available data. 
Normalos and Admins can see everything, but strangers should not be able to see students as search results!
Moreover, I will not only be able to exclude groups of people, but specific attributes on persons as well. Find a generic implementation for that, please. For this user story implement the example: Strangers should not be able to see students and  phone numbers attributes from persons in general. Excluded in search, map, detailpage, .....

### PR DESCRIPTION
**Pull request by treyfel**
Fixes #187 . Doesnt show students for not logged in users as search results.
Also adds two tests for this. Furthermore unwanted attributes can be added to the code.

### CODE REVIEW

File: app/views/search/index.html.erb [hunk 1]
```diff
@@ -5,57 +5,65 @@
...
+                  <div class='row'>
+                    <div class='col-2 d-flex flex-column'>
+                      <%= image_tag(result.image_or_placeholder, class: "picture-rounded md") %>
+                    </div>
+                    <div class="col d-flex flex-column my-2">
+                      <h5 class="search-title"><%= result.to_string %></h5>
+                      <div>
+                        <% result.displayed_tags.each do |tag| %>
+                          <% def prohibited_keys
+                               ["Telegram"]
```
Comment (treyfel) [inline app/views/search/index.html.erb:26]: This is where you can add attributes, that you don't want not signed in users to see.

**Review by lucasliebe — COMMENTED** [review]
I think we don't need the role attribute since as I know status is used for separating students and other members of the university. @FranzSw ?

**Review by lucasliebe — APPROVED** [review]

**Review by lucasliebe — CHANGES_REQUESTED** [review]
if tests run this can be approved

**Review by FranzSw — APPROVED** [review]
Exceptional code quality!

**Pull request merged**